src/Microsoft.AspNetCore.Authentication.Core/AuthenticationSchemeProvider.cs

```diff
@@ -144,8 +149,10 @@ public virtual void AddScheme(AuthenticationScheme scheme)
                 if (typeof(IAuthenticationRequestHandler).IsAssignableFrom(scheme.HandlerType))
                 {
                     _requestHandlers.Add(scheme);
+                    _requestHandlersCopy = _requestHandlers.ToList();
```

Comment:
Wouldn’t `.ToArray()` be a bit lighter here since we only use the result as an enumerable source? Or am I micro-optimizing too much?